### ISSUE TITLE
sympy.polys.polyerrors.PolificationFailed: Cannot construct polynomials exception using integrate 1.10.1 on Python 3.10.4

### ISSUE THREAD

**Issue by nasser1** [sympy/sympy#23704]
I do not know if this is known issue or fixed. If so, please feel free to close this. 

I am getting these exceptions using sympy 1.10.1 on Ubuntu 22.04 using Python 3.10.4. This is one example out of many

```
>python
Python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import *
>>> c , x, a, b = symbols('c x a b')
>>> integrate(log(x)/x**2/(c*x**2+b*x+a),x)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/integrals.py", line 1566, in integrate
    return integral.doit(**doit_flags)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/integrals.py", line 613, in doit
    antideriv = self._eval_integral(
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/integrals.py", line 956, in _eval_integral
    result, i = risch_integrate(f, x, separate_integral=True,
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/risch.py", line 1831, in risch_integrate
    ans, i, b = integrate_primitive(fa, fd, DE)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/risch.py", line 1460, in integrate_primitive
    q, i, b = integrate_primitive_polynomial(p, DE)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/risch.py", line 1409, in integrate_primitive_polynomial
    rv = limited_integrate(aa, ad, [(Dta, Dtb)], DE)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/prde.py", line 837, in limited_integrate
    h, A = param_rischDE(Fa, Fd, G, DE)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/integrals/prde.py", line 766, in param_rischDE
    M = Matrix([wl[:m] + wl[-v:] for wl in W])  # excise dj's.
  File "/home/me/.local/lib/python3.10/site-packages/sympy/polys/polymatrix.py", line 83, in __new__
    return cls.from_list(rows, cols, items, gens, ring)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/polys/polymatrix.py", line 106, in from_list
    items, info = parallel_poly_from_expr(items, gens, field=True)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/polys/polytools.py", line 4414, in parallel_poly_from_expr
    return _parallel_poly_from_expr(exprs, opt)
  File "/home/me/.local/lib/python3.10/site-packages/sympy/polys/polytools.py", line 4469, in _parallel_poly_from_expr
    raise PolificationFailed(opt, origs, exprs, True)
sympy.polys.polyerrors.PolificationFailed: Cannot construct polynomials from
>>>

```

**Comment by oscarbenjamin** [comment]
Regression since 63799649335f2d48c66bd533a8bdaeddc8973954.

I think the fix is
```diff
diff --git a/sympy/integrals/prde.py b/sympy/integrals/prde.py
index 9a6b74d..d818215 100644
--- a/sympy/integrals/prde.py
+++ b/sympy/integrals/prde.py
@@ -760,8 +760,14 @@ def param_rischDE(fa, fd, G, DE):
     # y = Sum(blk'*hk, (k, 1, v))/gamma, where k' = k + m + u.
 
     v = len(h)
-    M = Matrix([wl[:m] + wl[-v:] for wl in W])  # excise dj's.
+    shape = (len(W), m+v)
+    elements = [wl[:m] + wl[-v:] for wl in W] # excise dj's.
+    items = [e for row in elements for e in row]
+
+    # Need to set the shape in case W is empty
+    M = Matrix(*shape, items, DE.t)
     N = M.nullspace()
+
     # N = [n1, ..., ns] where the ni in Const(k)^(m + v) are column
     # vectors generating the space of linear relations between
     # c1, ..., cm, e1, ..., ev.
```

**Comment by eagleoflqj** [comment]
I guess every `rational_func_of_x * log(x)` should have a closed form result (with complex and polylog). This should be implemented in manualintegrate, but #23675 is a prerequisite.
